lib/edgeql_codegen.dart

```diff
@@ -336,7 +343,14 @@ _WalkCodecReturn _walkCodec(Codec codec, LibraryBuilder file,
     typeClass.methods.add((MethodBuilder()
           ..name = 'toMap'
           ..returns = Reference('Map<String, dynamic>')
-          ..body = literalMap(namesMap).returned.statement)
+          ..body = literalMap(namesMap.map((key, value) {
+            final ref = value['ref'] as Reference;
+            final mappable = value['mappable'] as bool;
+            return MapEntry(
+              literalString(key),
+              mappable ? ref.property('toMap').call([]) : ref,
```

Comment:
This also needs to handle fields that are nullable and/or a `List` of objects. Adding a couple of extra properties to the `namesMap` is probably the easiest way. Maybe something like this:
```
isList: child.typeRef.symbol == 'List',
isNullable: cardinality == Cardinality.atMostOne
```